data/src/main/java/io/jmix/data/impl/jpql/generator/SortJpqlGenerator.java

```diff
@@ -59,170 +59,164 @@ public String processQuery(String entityName, List<String> valueProperties, Stri
             return queryString;
         }
 
-        Map<Sort.Direction, List<Sort.Order>> directions = orders.stream()
-                .collect(Collectors.groupingBy(Sort.Order::getDirection));
-        if (directions.size() > 1) {
-            throw new UnsupportedOperationException("Sorting by multiple properties in different directions is not supported");
-        }
-        boolean asc = directions.keySet().iterator().next() == Sort.Direction.ASC;
+        Sort.Direction defaultSort = Sort.Direction.ASC;
 
-        List<String> sortExpressions = new ArrayList<>();
+        Map<String,Sort.Direction> sortExpressions = new LinkedHashMap<>();
```

Comment:
Everywhere in this file and other files use standard formatting, i.e. add a space between String and Sort.Direction in generics list: Map<String, Sort.Direction>

Reply:
fixed